scim-core/src/test/java/org/apache/directory/scim/core/repository/PatchHandlerTest.java

```diff
@@ -56,6 +56,46 @@ public PatchHandlerTest() {
     this.patchHandler = new DefaultPatchHandler(schemaRegistry);
   }
 
+  @Test
+  public void applyReplaceComplexAttribute() {
+    // setup existing user
+    Name existingName = new Name();
+    existingName.setFamilyName("Family Name");
+    existingName.setGivenName("Given Name");
+    ScimUser user = user();
+    user.setName(existingName);
+
+    // setup patch ops
+    Map<String, String> newName = Map.of("familyName", "New Family Name");
+    PatchOperation op = patchOperation(REPLACE, "name", newName);
+
+    //execute
+    ScimUser updatedUser = patchHandler.apply(user, List.of(op));
+    assertThat(updatedUser.getName().getFamilyName()).isEqualTo("New Family Name");
+    // assert that PATCH did not update fields not provided in PatchOperation
+    assertThat(updatedUser.getName().getGivenName()).isNotNull();
```

Comment:
nit: we may want to assert the original value here instead of null.

```suggestion
    assertThat(updatedUser.getName().getGivenName()).isEqualTo("Given Name");
```